buzzerComponent/buzzer.c

```diff
@@ -11,278 +25,289 @@
+/// @warning This must be one of the valid frequencies, otherwise the driver will reject it.
+static uint Frequency = 1024;
 
-/* To be more generic the ref needs to be a union of the returned dhub PushHandlerRef_t types
- * and PushHandler needs to be a union of different types
- */
-typedef struct actuator_DhResTypes_t {
-    const char        *OutputPath;
-    const char        *InputPath;
-    dhubIO_DataType_t dataType;
-    const char        *units;
-    void              (*OutputHandler)(double, double, void *);
-    void              *ref;     /* Need a union here */
-} actuator_DhResTypes_t;
-
-static void FreqConfigPushHandler (double timestamp, double freq, void *context);
-static void PeriodConfigPushHandler (double timestamp, double period, void *context);
-static void DcOnConfigPushHandler (double timestamp, double dcOnInterval, void *context);
-static void EnablePushHandler (double timestamp, bool enable, void *context);
-
-/* To be more generic the PushHandler dhub routine should be in the table */
-static actuator_DhResTypes_t dhubResources[] = {
-   {RES_PATH_ENABLE, NULL, DHUBIO_DATA_TYPE_BOOLEAN, "1/0", NULL},
-   {RES_PATH_FREQ_OUTPUT, RES_PATH_FREQ_INPUT, DHUBIO_DATA_TYPE_NUMERIC, "Hz",
-    FreqConfigPushHandler},
-   {RES_PATH_PERIOD_OUTPUT, RES_PATH_PERIOD_INPUT, DHUBIO_DATA_TYPE_NUMERIC, "s",
-    PeriodConfigPushHandler},
-   {RES_PATH_DC_ON_OUTPUT, RES_PATH_DC_ON_INPUT, DHUBIO_DATA_TYPE_NUMERIC, "%",
-    DcOnConfigPushHandler},
-};
-
-// The current frequency of square wave output by the rtc chip to the buzzer
-static int current_freq = 0;
-static int set_freq = 0;
-static FILE *freqFp = NULL;
-
-// The duty cycle percent of being on/off for the buzzer
-static int dc_on = 0;
-
-// The period of the duty cycle
-static int period = 0;
+// The on percentage of the buzzer on/off duty cycle (0 to 100).
+static uint DutyCycleOnPercent = 50;
+
+// The total number of milliseconds in the full duty cycle period (on + off).
+// Must be >= 100 and <= 3600000 (i.e. 1 hour).
+static uint PeriodMs = 2000;
 
 // The timer used to run the duty cycle
 static le_timer_Ref_t Timer = NULL;
-static uint32_t dc_on_timer = 0;
-static uint32_t dc_off_timer = 0;
-static bool buzzer_on = false;
 
-static void SetBuzzer(void)
+// true if the buzzer is currently on (buzzing).
+static bool BuzzerOn = false;
+
+//--------------------------------------------------------------------------------------------------
+/**
+ * Makes the buzzer sound at a given frequency in Hz.
+ */
+//--------------------------------------------------------------------------------------------------
+static void SetBuzzerHz
+(
+    uint freq    ///< The buzzer frequency, in Hz. Use BUZZER_OFF_FREQ to stop the buzzer.
+)
 {
-    if (freqFp == NULL)
+    // Path to the RTC CLKOUT control file in sysfs.
+    static const char BuzzerFreqPath[] = "/sys/bus/i2c/drivers/rtc-pcf85063/8-0051/clkout_freq";
+
+    static FILE *FreqFile = NULL;
```

Comment:
Is variable capitalization related to the lifetime or scope of the variable?

Reply:
I'm following this:
https://docs.legato.io/latest/ccodingStdsFormat.html#codingFormatNamingStaticVar

Reply:
Disregard the above comment.  It's clear after reading this: https://docs.legato.io/latest/ccodingStdsFormat.html#codingFormatNamingStackVar